torchvision/datasets/utils.py

```diff
@@ -42,6 +42,31 @@ def check_integrity(fpath: str, md5: Optional[str] = None) -> bool:
     return check_md5(fpath, md5)
 
 
+def _get_google_drive_file_id(url: str) -> Optional[str]:
+    import re
+
+    pattern = re.compile(
+        '(?:(?:(?:http)s?://)?(?:(?:docs)|(?:drive))\\.(?:google)\\.(?:com))(?P<request>.*)',
+        re.IGNORECASE
+    )
+    match = pattern.match(url)
+
+    file_id = None
```

Comment:
Instead of setting a default and maybe overwriting it later you could simply insert a guard clause

```suggestion
    if match is None:
        return None
```

This also saves one level of indentation.